src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1870,7 +1522,296 @@ public void SetTransformer(string transformer)
+            var localStringsCache = _implicitStringsCache;
+            if (localStringsCache.TryGetValue(type, out value))
+                return value;
+
+            var methodInfo = type.GetMethod("op_Implicit", new[] { type });
+
+            if (methodInfo == null || methodInfo.ReturnType != typeof(string))
+            {
+                _implicitStringsCache = new Dictionary<Type, Func<object, string>>(localStringsCache)
+                {
+                    {type, null}
+                };
+                return null;
+            }
+
+            var arg = Expression.Parameter(typeof(object), "self");
+
+            var func = (Func<object, string>)Expression.Lambda(Expression.Call(methodInfo, Expression.Convert(arg, type)), arg).Compile();
+
+            _implicitStringsCache = new Dictionary<Type, Func<object, string>>(localStringsCache)
+            {
+                {type, func}
+            };
+            return func;
+        }
+
+        private object TransformValue(WhereParams whereParams)
+        {
+            if (whereParams.Value == null)
+                return null;
+            if (Equals(whereParams.Value, string.Empty))
+                return string.Empty;
+
+            var type = whereParams.Value.GetType().GetNonNullableType();
+
+            if (type == typeof(DateTime) || type == typeof(DateTimeOffset))
+                return whereParams.Value;
+
+            if (type == typeof(string))
+            {
+                var valueAsString = (string)whereParams.Value;
+                return whereParams.IsAnalyzed ? valueAsString : string.Concat("[[", valueAsString, "]]");
+            }
+
+            if (type == typeof(int))
+                return whereParams.Value;
+            if (type == typeof(long))
+                return whereParams.Value;
+            if (type == typeof(decimal))
+                return whereParams.Value;
+            if (type == typeof(double))
+                return whereParams.Value;
+            if (whereParams.Value is TimeSpan)
+                return whereParams.Value;
+            if (whereParams.Value is float)
+                return whereParams.Value;
+            if (whereParams.Value is string)
+                return whereParams.Value;
+
+            if (whereParams.Value is ValueType)
+                return Convert.ToString(whereParams.Value, CultureInfo.InvariantCulture);
+
+            var result = GetImplicitStringConversion(whereParams.Value.GetType());
+            if (result != null)
+                return result(whereParams.Value);
+
+            return whereParams.Value;
+
+            //if (whereParams.Value is ValueType)
+            //    return RavenQuery.Escape(Convert.ToString(whereParams.Value, CultureInfo.InvariantCulture), false, true);
+
+            //string strVal;
+            //if (_conventions.TryConvertValueForQuery(whereParams.FieldName, whereParams.Value, QueryValueConvertionType.Range, out strVal))
```

Comment:
Probably need a TODO here to remember to restore it.
Then again, our rules for types are basically long, double, string, bool or null, so we just need to be able to convert it to a string.